### PR TITLE
test: switch tooltip tabstop tests to react testing library

### PR DESCRIPTION
**Pull request by lsprr**
Issue #1159

This PR switches the Tooltip Tabstop tests to React Testing Library.

Validation steps

- Run yarn test
- See that there are coverage percentages in the table
- Verify that the tests are converted properly

### CODE REVIEW

File: packages/react/src/components/TooltipTabstop/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+test('should render without errors', () => {
+  render(<TooltipTabstop tooltip="World">Hello</TooltipTabstop>);
+  expect(screen.getByText('Hello')).toBeInTheDocument();
+});
+
+test('should display tooltip on hover', async () => {
+  render(<TooltipTabstop tooltip="World">Hello</TooltipTabstop>);
+
+  await userEvent.hover(screen.getByText('Hello'));
+  expect(screen.getByText('World')).toBeInTheDocument();
```
Comment (scurker) [inline packages/react/src/components/TooltipTabstop/index.test.tsx:16]: If you look at how we structure the [tooltip tests](https://github.com/dequelabs/cauldron/blob/0efb20782bc2943a438d9810d065cdf0af93efd1/packages/react/src/components/Tooltip/Tooltip.test.tsx#L67-L73) we likely want these to be very similar to that: 

```suggestion
  expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
  await fireEvent.focusIn(screen.getByRole('button'));
  expect(screen.queryByRole('tooltip')).toBeInTheDocument();
```

Comment (scurker) [reply packages/react/src/components/TooltipTabstop/index.test.tsx:16]: Additionally, we want to validate the accessible description as well:

```tsx
expect(
  screen.getByRole('button', { name: 'button' })
).toHaveAccessibleDescription('World');
```

**Review by scurker — CHANGES_REQUESTED** [review]
Just some minor changes needed.

**Comment by lsprr** [comment]
@scruker, thank you for the feedback! I will review other pull requests to ensure everything is correct.

**Review by scurker — APPROVED** [review]
🚨 Reviewed for Security 🚨

**Pull request merged**